test/verifier.spec.js

```diff
@@ -873,6 +873,46 @@ test('it validates allowed claims values using equality when appropriate', t =>
   t.end()
 })
 
+test('it validates whether a required claim is present in the payload or not', t => {
+  // Token payload: { "iss": "ISS"}
+  const token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJJU1MifQ.FKjJd2A-T8ufN7Y0LpjMR23P7CwEQ3Y-LBIYd2Vh_Rs'
+
+  t.strictSame(
+    verify(
+      token,
+      { allowedIss: 'ISS', requiredClaims: ['iss'] }
+    ),
+    { iss: 'ISS' }
+  )
+
+  t.throws(
+    () => {
+      return verify(
+        token,
+        { allowedSub: 'SUB', requiredClaims: ['sub'] }
+      )
+    },
+    { message: 'The sub claim is required.' }
+  )
+
+  t.end()
+})
```

Comment:
minor: I would split this test in two different tests. one for the successful behavior, another for the failure behavior